### ISSUE TITLE
ClassLoaderHandlers are not found when gradle + ant is used together

### ISSUE THREAD

**Issue by vaclavnemec** [lukehutch/fast-classpath-scanner#44]
We have a gradle build which is using old ant tasks from our old build system (gradle has support for ant).

The next code snippet is problematic. No handler is found.
```
private static ServiceLoader<ClassLoaderHandler> classLoaderHandlerLoader = ServiceLoader
            .load(ClassLoaderHandler.class);
```
The root cause seems to be different class loader. ServiceLoader uses `Thread.currentThread().getContextClassLoader()`

It would be very useful to pass classloader as an parameter of `FastClasspathScanner` or get the classloader from caller.

**Comment by lukehutch** [comment]
Thanks for the report, interesting. ServiceLoader is supposed to be supported by all classloaders, as far as I know -- it is a Java platform feature. The fully-qualified classnames of the handlers are listed here:

https://github.com/lukehutch/fast-classpath-scanner/blob/master/src/main/resources/META-INF/services/io.github.lukehutch.fastclasspathscanner.scanner.classloaderhandler.ClassLoaderHandler

Somehow that META-INF directory is not being properly located in your setup. I have no knowledge of gradle, and very little knowledge of ant. Do you know if ServiceLoader is even supposed to work with gradle?

I don't know how to duplicate this, is there any chance you could please produce a minimal testcase, with instructions for building/running it?

**Comment by vaclavnemec** [comment]
please see https://github.com/vaclavnemec/file-classpath-scanner-gradle-ant-issue

its autogenerated gradle project with some changes is build.gradle and with ant task MyTask.groovy which creates FastClasspathScanner and writes some logs. I hope there are all information in readme

**Comment by lukehutch** [comment]
Thanks for the minimal test case, very helpful, I can reproduce this.

The two classloaders I can see are the following.

```
sun.misc.Launcher$AppClassLoader
org.gradle.internal.classloader.MutableURLClassLoader
```

I need to do more investigation to figure out how to get classpath elements out of these classloaders. 

I also need to figure out why ServiceLoaders don't work in Gradle.

**Comment by vaclavnemec** [comment]
Please see ServiceLoader source

```
public static <S> ServiceLoader<S> load(Class<S> service) {
        ClassLoader cl = Thread.currentThread().getContextClassLoader();
        return ServiceLoader.load(service, cl);
    }
```

its using `Thread.currentThread().getContextClassLoader()` - the returned class loader probably does not see [META-INF/services/io.github.........ClassLoaderHandler](https://github.com/lukehutch/fast-classpath-scanner/blob/master/src/main/resources/META-INF/services/io.github.lukehutch.fastclasspathscanner.classpath.classloaderhandler.ClassLoaderHandler)

**Comment by lukehutch** [comment]
Hi, I just pushed some fixes that should solve this problem. Your code had the correct idea for the fix (specify a specific class' classloader as the second parameter to ServiceLoader), thanks for the suggestion as to how to fix this, I don't know if I would have figured this out! I would be grateful if you could please test the changes. 

Note that since 1.9.18, I renamed some other packages, so for your test code, you need to rename the imports from ".scanner." to ".classpath."  Also, one line of your test code depends on a field that no longer exists in this version, you can just delete that line.  Or you could just try this directly with the code you were trying to get working, hopefully it fixes it. (Your test case works for me.)

**Comment by lukehutch** [comment]
Reopening until confirmed.

**Comment by vaclavnemec** [comment]
Thank you very much it is working now.

**Comment by lukehutch** [comment]
Great, thanks so much again for the report, the test case, the solution, and the followup testing! Will push out a new release in a couple of days once I finish some other changes.

**Comment by lukehutch** [comment]
Fixed in 1.9.19.
